### PR TITLE
Support two apiversion for container service

### PR DESCRIPTION
**Pull request by zqingqing1**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by annatisch** [comment]
Adding @ravbhatnagar for ARM review of new API version.

@zqingqing1 - the changes you made to the readme have removed the OpenShiftManagedClusters spec - was this intentional?

**Comment by zqingqing1** [comment]
@annatisch , thanks for reminding, it should be included.

**Comment by zqingqing1** [comment]
Hi, @annatisch @ravbhatnagar please help to review :)

**Comment by annatisch** [comment]
@zqingqing1 - please address the model validation errors in the CI:
```
error:
operationId: ManagedClusters_List
scenario: List Managed Clusters
source: response
responseCode: '200'
errorCode: OBJECT_ADDITIONAL_PROPERTIES
errorDetails:
  message: 'Additional properties not allowed: storageProfile'

error:
operationId: ManagedClusters_ListByResourceGroup
scenario: Get Managed Clusters by Resource Group
source: response
responseCode: '200'
errorCode: OBJECT_ADDITIONAL_PROPERTIES
errorDetails:
  message: 'Additional properties not allowed: storageProfile'

error:
operationId: ManagedClusters_Get
scenario: Get Managed Cluster
source: response
responseCode: '200'
errorCode: OBJECT_ADDITIONAL_PROPERTIES
errorDetails:
  message: 'Additional properties not allowed: storageProfile'

error:
operationId: ManagedClusters_CreateOrUpdate
scenario: Create/Update Managed Cluster
source: response
responseCode: '200'
errorCode: OBJECT_ADDITIONAL_PROPERTIES
errorDetails:
  message: 'Additional properties not allowed: storageProfile'

error:
operationId: ManagedClusters_CreateOrUpdate
scenario: Create/Update Managed Cluster
source: response
responseCode: '201'
errorCode: OBJECT_ADDITIONAL_PROPERTIES
errorDetails:
  message: 'Additional properties not allowed: storageProfile'

error:
operationId: ManagedClusters_UpdateTags
scenario: Update Managed Cluster Tags
source: response
responseCode: '200'
errorCode: OBJECT_ADDITIONAL_PROPERTIES
errorDetails:
  message: 'Additional properties not allowed: storageProfile'
```

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (KrisBash) [inline specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: This appears to be deprecation of an API from a stable version. Do you have a deprecation policy in place? Has this been communicated?

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comments on API deprecation

**Comment by zqingqing1** [comment]
@KrisBash , I just saw your comments about api deprecation from email. Not sure why I cannot see the comments here. Anyway, for your question:
> @@ -179,58 +179,6 @@
         }
       }
     },
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/accessProfiles/{roleName}/listCredential": {
This appears to be deprecation of an API from a stable version. Do you have a deprecation policy in place? Has this been communicated?

No, right now we don't have the deprecation policy. This is a long-term work item, and needs extra discussion with PM team. For now, we just keep it there.

**Comment by KrisBash** [comment]
As of latest commit, looks good to me from ARM

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Comment by zqingqing1** [comment]
@annatisch @KrisBash @jhendrixMSFT, I am blocked by this. Can anyone of you help to merge the PR?  Thanks in advance.

**Comment by annatisch** [comment]
@zqingqing1 - the PR name implies that you wish to support two api versions - yet the readme changes do no reflect this. What is your intention here?

**Comment by zqingqing1** [comment]
Synced up with @lmazuel before, after this PR gets merged, he will help to support both apiversion for Azure-SDK-for-Python.

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (annatisch) [inline specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: A quick comment on the use of "defaults".

It's important to note that how "defaults" are handled in a generated SDK can vary from language to language.
Some languages will send the default values in the request body (regardless of whether it's set by the user) and other languages will leave these fields in the request as empty, and use the default purely to document the service behavior.

So please make sure that your service API will handle both scenarios the same way - i.e. it will treat a request with an empty field and a request with a default value exactly the same.

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (annatisch) [inline specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: Typo: "Minimum"

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (annatisch) [inline specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: Typo: "Maximum"

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (annatisch) [inline specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: Typo: "Whether"

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (annatisch) [inline specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: I see "storageProfile" still exists here, whereas it has been removed from "ManagedClusterAgentPoolProfile" - this is as you intended?

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (annatisch) [inline specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: I see you've used a Boolean for this field - we try to encourage teams to use string/enums instead of Boolean as they are more descriptive and allow for non-breaking changes if you later wish to add more states.
The use of a Boolean here is actually [generating an ARM warning in the CI](https://github.com/Azure/azure-rest-api-specs/pull/4237/checks?check_run_id=24725712#user-content-ARM-Warnings) - is it possible to change this or is the API now live?

If not - then I wont block on it, but bare in mind for future API versions :)

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (annatisch) [inline specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: This was removed in a PR a month ago - I think you need to rebase to get the latest changes.

**Review by annatisch — CHANGES_REQUESTED** [review]
Mostly minor things :)

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (zqingqing1) [reply specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: Yes, it is for ContainerServiceMasterProfile.

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (zqingqing1) [reply specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: Oops, corrected it to right word

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (zqingqing1) [reply specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: From our Service Api, this field is read-only. So I will remove the "default field"

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (zqingqing1) [reply specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: It only has 2 states, so boolean type is enough for the case.

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json
Comment (annatisch) [reply specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: Thanks @zqingqing1 - though it it's read-only you can mark it as "readOnly": true in the "ManagedClusterAgentPoolProfile"

**Comment by annatisch** [comment]
Thanks @zqingqing1 - please address my earlier comment on breaking changes/major SDK version bump :)

**Comment by zqingqing1** [comment]
@annatisch , yes.

**Comment by annatisch** [comment]
Thanks @zqingqing1 - This will result in your new public API not being available in other SDKs.

Adding multi-api support only affects Python - whereas reverting the tag will make the public API not available in other languages. If this API is live and public, the SDK generation tags should reflect this and then you can work further with @lmazuel around the specifics of the Python release.

**Review by annatisch — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Pull request merged**
